Source/Core/InputCommon/ControllerEmu/ControllerEmu.h

```diff
@@ -37,6 +38,23 @@ struct TwoPointCalibration
   TwoPointCalibration() = default;
   TwoPointCalibration(const T& zero_, const T& max_) : zero{zero_}, max{max_} {}
 
+  constexpr bool IsSane() const
+  {
+    if constexpr (std::is_arithmetic_v<T>)
+    {
+      return max != zero;
+    }
+    else
+    {
+      for (size_t i = 0; i != std::size(zero.data); ++i)
```

Comment:
This can likely be a `std::all_of`

Reply:
I'm indexing two arrays. I don't think it can be a clean `std::all_of` but I'll change it to an `std::equal`.